lib/hammer_cli/abstract.rb

```diff
@@ -233,10 +232,12 @@ def self.option(switches, type, description, opts = {}, &block)
     end
 
     def all_options
-      self.class.recognised_options.inject({}) do |h, opt|
-        h[opt.attribute_name] = send(opt.read_method)
-        h
+      self.class.recognised_options.inject({}) do |hash, opt|
+        hash[opt.attribute_name] = send(opt.read_method)
+        hash[opt.attribute_name] = context[:defaults].get_defaults(opt.attribute_name) if hash[opt.attribute_name].nil? && context[:defaults]
```

Comment:
Can you keep the logging of the default being used that was present in previous version. It can make bug hunting easier in the future.